lib/p2p/packets/Packet.ts

```diff
@@ -1,11 +1,36 @@
 import PacketType from './PacketType';
+import CryptoJS from 'crypto-js';
+import SHA256 from 'crypto-js/sha256';
+
+type PacketHeader = {
+  hash?: string;
```

Comment:
Should we call this `resHash` or something? These fields are a bit confusing to me, especially when it's not clear whether a packet is outgoing/ingoing, so making this as clear as possible how they are being used (maybe with some tsdoc) would be helpful.

Reply:
Ok so `hash` is the hash of the body of the packet, and reqHash is the hash of the packet which we are responding to, or null if we're not responding to a packet. That's fine, I just think comments would be good.